app/assets/javascripts/Components/Modals/send_lti_grades_modal.jsx

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import Modal from "react-modal";
+
+class LtiGradeModal extends React.Component {
+  static defaultProps = {
+    override: false,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      deployments: this.props.lti_deployments,
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentDidMount() {
+    Modal.setAppElement("body");
+    const deploymentsMapped = this.state.deployments.map(deployment => ({
```

Comment:
Are you sure you want this to run in componentDidMount instead of componentDidUpdate? What happens if the props change while the component has already been mounted? Is that a possibility?

Reply:
Right now, the props are never modified, and shouldn't be modified, but you're right - it makes sense to do it in componentDidUpdate

Reply:
After thinking about it, I think this does need to be in `componentDidMount`, because it sets the initial state which is then modified by user actions (checking or unchecking the box for each lti deployment).

Reply:
Ok then it should be set in the constructor as an initial value in the state